okhttp/src/main/java/okhttp3/RealCall.java

```diff
@@ -72,20 +81,32 @@ static RealCall newRealCall(OkHttpClient client, Request originalRequest, boolea
       executed = true;
     }
     captureCallStackTrace();
+    timeout.enter();
     eventListener.callStart(this);
     try {
       client.dispatcher().executed(this);
       Response result = getResponseWithInterceptorChain();
       if (result == null) throw new IOException("Canceled");
       return result;
     } catch (IOException e) {
+      e = timeoutExit(e);
       eventListener.callFailed(this, e);
       throw e;
     } finally {
       client.dispatcher().finished(this);
     }
   }
 
+  IOException timeoutExit(@Nullable IOException cause) {
+    if (!timeout.exit()) return cause;
+
+    InterruptedIOException e = new InterruptedIOException("timeout");
```

Comment:
Is it useful to include the timeout time in the exception? Would flag misconfiguration to potential bug reporters.

Reply:
Ick, this is awkward because we don’t have a good toString() on timeout. Will fix in follow-up.

Reply:
(and elsewhere that we do this!)

Reply:
https://github.com/square/okhttp/issues/4367